src/index.js

```diff
@@ -329,17 +344,23 @@ const reduceRules = (rules, attributes) => {
  *
  * @param {object} policy the policy to evaluate
  * @param {object} attributes the attributes to use for the evaluation
+ * @param {object} options optional function config
+ * @param {array} options.eagerTargets optional list of attribute paths that
+ * should be eagerly evaluated when reducing the policy. Eager evaluation makes
+ * sure that a rule with a known target will be inverted and replaced with the
+ * known value in-line.
```

Comment:
This looks like the API we talked about, nice.

I'm hesitant to comment it, but after looking at this, I had an idea for a _slightly_ different API that I think might serve a bit better -- curious what you think:

Basically, instead of `eagerTargets`, support an `inlineAttributes` option, which is just an attribute map, but for inlining.

```typescript
const reduced = reduce(
  policy,
  {
    // won't be inlined
    user: { id, groups }
  },
  {
    inlineAttributes: {
      // will be inlined
      user: {
        customAttributes: { ... }
      }
    }
  }
);
```

I think this API would provide two benefits:
1. The existing API relies on startsWith, which seems like it could theoretically lead to some weird behavior, e.g.:

```typescript
const reduced = reduce(
  policy,
  {
    user: {
      // this will match
      customAttributesOther: {...}
    }
  },
  {
    eagerTargets: [
      'user.customAttributes'
    ]
  }
);

```

There are definitely other ways we could get around that issue without changing the API -- just one solution.

2. It clearly states the "inlining" behavior in the name.

Curious what you think. Only reason I thought it was worth discussing was that this package is so widespread.

Reply:
Actually, I'm not sure the inlineAttributes API is better, since it allows for conflicts:

```typescript
const reduced = reduce(
  policy,
  {
    user: {
      id: 'foo'
    }
  },
  {
    inlineAttributes: {
      user: {
        id: 'bar'
      }
    }
  }
);
```

So, revising my feedback to these items:
- Should we rename `eagerTargets` to `inlineTargets`? I'm wondering whether that more clearly communicates the behavior.


- One weird thing we saw in debugging was that references to objects were being inlined (`user` in `connect-service`). Should we patch our logic to only inline values if they are arrays, numbers, or strings?

Reply:
Renaming to that sounds good. I'll also take care of fixing the issue that can come up by relying on `startsWith` that you brought up.